src/Microsoft.IdentityModel.Tokens/Validation/TokenValidationResult.cs

```diff
@@ -136,6 +136,8 @@ internal ClaimsIdentity ClaimsIdentityNoLocking
                     if (_validationParameters != null && SecurityToken != null && _tokenHandler != null && Issuer != null)
                     {
                         _claimsIdentity = _tokenHandler.CreateClaimsIdentityInternal(SecurityToken, _validationParameters, Issuer);
+                        if (_claimsIdentity is not CaseSensitiveClaimsIdentity && !AppContextSwitches.UseClaimsIdentityType())
```

Comment:
Why do you want to keep the above line that creates the potentially case-insensitive version? Is there a side effect you'd like to retain?

Reply:
Same behavior as with TVP.CreateClaimsIdentity; `_tokenHandler` can be derived and return normal ClaimsIdentity, which we'd honor. Although, we may not need this check here, since `_claimsIdentity` here will always probably be CsClaimsIdentity when switch is set.